subtests/docker_cli/run_cgroups/cgroup_parent.py

```diff
@@ -0,0 +1,103 @@
+"""
+Tests for bz1320302 - odd behavior of --cgroup-parent option
+
+Operational Summary
+----------------------
+
+#. Start container with --cgroup-parent=X, for various values of X
+#. run container command 'tail -n1 /proc/1/cgroup'
+#. Compare to expected result
+"""
+
+
+import os.path
+import re
+from autotest.client import utils
+from dockertest.dockercmd import DockerCmd
+from dockertest.images import DockerImage
+from dockertest.output import OutputGood
+from dockertest.xceptions import DockerTestError, DockerTestFail
+from cgroups_base import cgroups_base
+
+
+class cgroup_parent_base(cgroups_base):
+    def _setup(self, cg_parent, expected_path, expected_stderr):
+        """
+        Set up test parameters. Our caller gives us a value to use
+        for --cgroup-parent, and what to expect in stdout & stderr.
+        We replace the format strings {rand1} and {rand2} with two
+        pseudorandom strings, and {cid} with the container ID (once
+        we have it).
+        """
+        self.sub_stuff['rand1'] = utils.generate_random_string(8)
+        self.sub_stuff['rand2'] = utils.generate_random_string(8)
+        self.sub_stuff['cgroup_parent'] = cg_parent.format(**self.sub_stuff)
+        self.sub_stuff['expected_path'] = expected_path
+        self.sub_stuff['expected_stderr'] = expected_stderr
```

Comment:
BTW (in case you were wondering), test-data is kept in ``stuff`` and ``sub_stuff`` (they could be merged to use same name) is to keep the runtime namespace separated from the class and API.  It's just a simple way to avoid smashed toes (which has happened, along days spent tracking down ``-/_`` mix-ups and speling mistakes).